### PR TITLE
Migrate to pydantic 2.0+

### LINKED ISSUE #72: Bump to pydantic 2.0+
Lets move to pydantic 2, so that downstream (e.g) ASAPDiscovery can begin migrating.  

I say lets just move to raw pydantic 2, no need for a shim, we should just migrate properly once.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #72 

## Status
- [X] Ready to go

### CODE REVIEW

**Comment by hmacdope** [comment]
@kaminow could I grab a review here? This should allow us to migrate nicely and unblock downstream, but please double check everything still works as expected for configs.

File: mtenn/config.py [hunk 1]
```diff
@@ -577,14 +578,16 @@ def massage_into_lists(cls, values) -> GATModelConfig:
...
+            self.__dict__.update(values)
+            
 
         # Adjust any length 1 list to be the right length
         for p, list_len in list_lens.items():
             if list_len == 1:
                 values[p] = values[p] * num_layers
 
-        return values
+        return self.__dict__.update(values)
```
Comment (kaminow) [inline mtenn/config.py:590]: I don't think `update` actually returns anything
```suggestion
        self.__dict__.update(values)
        return self.__dict__
```

**Review by kaminow — APPROVED** [review]
one thing to address but otherwise looks good. thanks for doing this!

File: mtenn/config.py (hunk 1 above)
Comment (hmacdope) [reply mtenn/config.py:590]: Now that you mention it we should return `self`

Comment (hmacdope) [reply mtenn/config.py]: this is due to how the `after` mode validator works in pydantic 2.

**Pull request merged**